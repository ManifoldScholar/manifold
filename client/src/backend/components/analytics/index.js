import Block from "./Block";
import Grid from "./Grid";
import RangePicker from "./parts/RangePicker";
import ComposedAnalytics from "./composed";
import AnalyticsFactory from "./Factory";

export default Block;
export { Grid, RangePicker, AnalyticsFactory, ComposedAnalytics };
