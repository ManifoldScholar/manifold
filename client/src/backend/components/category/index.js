import Form from "./Form";
import List from "./List";

export default {
  Form,
  List
};
