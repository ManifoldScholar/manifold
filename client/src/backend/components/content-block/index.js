import Builder from "./Builder";
import TypeForm from "./TypeForm";
import DrawerHeader from "./DrawerHeader";

export default {
  Builder,
  TypeForm,
  DrawerHeader
};
