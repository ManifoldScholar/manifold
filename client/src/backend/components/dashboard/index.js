import Activity from "./Activity";
import Notifications from "./Notifications";
import Counts from "./Counts";
import Analytics from "./Analytics";

export default {
  Activity,
  Notifications,
  Counts,
  Analytics
};
