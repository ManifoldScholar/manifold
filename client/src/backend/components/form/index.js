import BaseInput from "./BaseInput";
import BelongsTo from "./BelongsTo";
import CodeArea from "./CodeArea";
import AttributeMap from "./AttributeMap";
import Date from "./Date";
import DatePicker from "./DatePicker";
import FieldGroup from "./FieldGroup";
import GeneratedPasswordInput from "./GeneratedPasswordInput";
import Header from "./Header";
import HasMany from "./HasMany";
import Hidden from "./Hidden";
import Instructions from "./Instructions";
import MaskedTextInput from "./MaskedTextInput";
import NumberInput from "./NumberInput";
import OptionsList from "./OptionsList";
import Radios from "./Radios";
import Save from "./Save";
import Select from "./Select";
import setter from "./setter";
import Switch from "./Switch";
import SwitchArray from "./SwitchArray";
import TagList from "./TagList";
import TextArea from "./TextArea";
import TextInput from "./TextInput";
import TusUpload from "./TusUpload";
import Upload from "./Upload";

export default {
  BaseInput,
  BelongsTo,
  CodeArea,
  AttributeMap,
  Date,
  DatePicker,
  FieldGroup,
  GeneratedPasswordInput,
  HasMany,
  Header,
  Hidden,
  Instructions,
  MaskedTextInput,
  NumberInput,
  OptionsList,
  Radios,
  Save,
  Select,
  setter,
  Switch,
  SwitchArray,
  TagList,
  TextArea,
  TextInput,
  TusUpload,
  Upload
};
