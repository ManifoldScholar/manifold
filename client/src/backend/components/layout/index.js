import GlobalHeader from "./GlobalHeader";
import BackendPanel from "./BackendPanel";
import PageHeader from "./PageHeader";
import DrawerHeader from "./DrawerHeader";
import SecondaryNav from "./SecondaryNav";

export default {
  GlobalHeader,
  PageHeader,
  BackendPanel,
  DrawerHeader,
  SecondaryNav
};
