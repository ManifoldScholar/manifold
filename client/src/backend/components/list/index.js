import Orderable from "./Orderable";
import SimpleList from "./SimpleList";
import Searchable from "./Searchable";

export default {
  Orderable,
  SimpleList,
  Searchable
};
