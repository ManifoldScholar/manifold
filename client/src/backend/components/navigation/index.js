import DetailHeader from "./DetailHeader";
import DrawerHeader from "./DrawerHeader";
import Dropdown from "./Dropdown";
import Secondary from "./Secondary";

export default {
  DetailHeader,
  DrawerHeader,
  Dropdown,
  Secondary
};
