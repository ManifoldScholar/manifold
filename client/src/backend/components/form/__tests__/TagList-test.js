import React from "react";
import renderer from "react-test-renderer";
import { Form } from "components/backend";
import { Provider } from "react-redux";
import createStore from "store/createStore";
const store = createStore();

describe("Backend.Form.TagList component", () => {
  const element = (
    <Provider store={store}>
      <Form.TagList label="Tags" name="attributes[tags]" />
    </Provider>
  );

  it("renders correctly", () => {
    const component = renderer.create(element);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
