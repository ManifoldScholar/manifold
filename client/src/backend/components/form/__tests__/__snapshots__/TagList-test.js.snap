// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backend.Form.TagList component renders correctly 1`] = `
<div
  className="form-input"
  style={Object {}}
>
  <label
    className=""
    htmlFor="attributes[tags]-tag-list-1"
  >
    Tags
  </label>
  <div
    className="input-predictive"
  >
    <div
      className="input-predictive__input"
    >
      <input
        aria-describedby={null}
        aria-label="Enter a Tag"
        className="input-predictive__text-input text-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        placeholder="Enter a Tag"
        type="text"
        value=""
      />
      <button
        aria-hidden="true"
        className="input-predictive__button"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="manicon-svg input-predictive__icon svg-icon--plus16"
          fill="currentColor"
          height={20}
          viewBox="0 0 16 16"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.01435,8.5138 L2.014482,7.5138 L14.01435,7.5149 L14.014218,8.5149 L2.01435,8.5138 Z M8.5149,14.014218 L7.5149,14.01435 L7.5138,2.014482 L8.5138,2.01435 L8.5149,14.014218 Z"
          />
        </svg>
      </button>
    </div>
    <nav
      className="input-predictive__results-wrapper"
    >
      <ul
        className="input-predictive__results"
      />
    </nav>
  </div>
  <div
    className="has-many-list"
  >
    <ul
      className="bucket empty"
    >
      <div
        className="placeholder"
      >
        None added
      </div>
    </ul>
  </div>
</div>
`;
